### PR TITLE
Use POST instead of GET and validate CSRF token

### PR DESCRIPTION
**Pull request by jakubroztocil**

### CODE REVIEW

**Comment by jakubroztocil** [comment]
#22

File: loginas/templates/loginas/change_form.html [hunk 1]
```diff
@@ -6,7 +6,18 @@
...
     <li><a href="history/" class="historylink">History</a></li>
-    <li><a href="{% url "loginas-user-login" object_id %}">Log in as user</a></li>
+    <li><a id="loginas-link" href="#">Log in as user</a></li>
   </ul>
+
+  <form method="post" id="loginas-form" action="{% url 'loginas-user-login' object_id %}" stype="display:none">
+    {% csrf_token %}
+  </form>
+  <script>
+    django.jQuery('#loginas-link').on('click', function() {
```
Comment (skorokithakis) [inline loginas/templates/loginas/change_form.html]: Does this catch keyboard events?

**Comment by skorokithakis** [comment]
It looks like tests are broken for any version of Django less than 1.7, is that fixable?

File: loginas/templates/loginas/change_form.html (hunk 1 above)
Comment (jakubroztocil) [inline loginas/templates/loginas/change_form.html]: Yes, it does (http://jsfiddle.net/5hcby28s/).

Comment (skorokithakis) [inline loginas/templates/loginas/change_form.html]: Fantastic, thank you.

**Comment by jakubroztocil** [comment]
Not sure whether you've noticed, but I've added a commit fixing the tests on all the supported Django/Python combinations.

**Comment by skorokithakis** [comment]
Hmm, two combinations failed, I'll look into it. Apparently Django 1.7 doesn't support 2.6...

**Comment by skorokithakis** [comment]
Thanks! Odd that this is failing, where's it getting the Django 1.7.1 from?

**Pull request merged**